modules/tests/test_ref_modules/test_ref_modules.cpp

```diff
@@ -575,5 +575,493 @@ TEST_F(RefModulesTest, UpdateDevicePower)
+
+    // Checking dimension of classifier
+    {
+        auto signalDimension = classifierSignalDescription.getDimensions()[0];
+        auto labels = signalDimension.getLabels();
+        Float expectedMark = classifierInputRange.getLowValue();
+        for (const auto& mark : labels)
+        {
+            ASSERT_EQ(static_cast<Float>(mark), expectedMark);
+            expectedMark += classCount;
+        }
+        ASSERT_EQ(labels[labels.getCount() - 1], classifierInputRange.getHighValue());
+    }
+
+    const auto classifierDomainSignal = classifierSignal.getDomainSignal();
+    const auto classifierDomainSignalDescription = classifierDomainSignal.getDescriptor();
+
+    const auto deviceDomainDescriptor = deviceSignal.getDomainSignal().getDescriptor();
+
+    const auto deviceDomainRule = deviceDomainDescriptor.getRule();
+    const auto classifierDomainRule = classifierDomainSignalDescription.getRule();
+
+    // Check Linear Rule
+    if (deviceDomainRule.getType() == DataRuleType::Linear)
+    {
+        ASSERT_EQ(deviceDomainRule.getType(), classifierDomainRule.getType());
+        ASSERT_EQ(classifierDomainSignalDescription.getTickResolution(), deviceDomainDescriptor.getTickResolution());
+
+        auto deviceResolution = deviceDomainDescriptor.getTickResolution() * Int(1000);
+
+        auto classifierDelta = classifierDomainRule.getParameters().get("delta");
+        auto expectedClassifierDelata = static_cast<Float>(blockSize) / static_cast<Float>(deviceResolution);
+
+        ASSERT_EQ(classifierDelta, expectedClassifierDelata);
+    }
+    else
+    {
+        ASSERT_EQ(classifierDomainRule.getType(), DataRuleType::Explicit);
+    }
+}
+
+TEST_F(RefModulesTest, ClassifierBlackBox)
+{
+    using inputSignalType = Int;
+    SampleType inputSignalSampleType = SampleType::Int64;
+    using outputSignalType = Float;
+
+    auto inputSignalRange = Range(0, 20);
+    bool inputSignalSync = true;
+
+    InputStubSignal input;
+    input.setUp(inputSignalSampleType, inputSignalRange, inputSignalSync);
+    auto instance = input.getInstance();
+    auto inputSignal = input.getSignal();
+
+    SizeT blockSizeMs = 5;
+    SizeT classCount = 1;
+    SizeT inputPacketCount = blockSizeMs;
+    SizeT outputPacketCount = inputPacketCount / blockSizeMs;
+
+    auto classifierFb = instance.addFunctionBlock("ref_fb_module_classifier");
+    classifierFb.setPropertyValue("UseCustomClasses", false);
+    classifierFb.setPropertyValue("BlockSize", blockSizeMs);
+    classifierFb.setPropertyValue("ClassCount", classCount);
+
+    const auto classifierPort = classifierFb.getInputPorts()[0];
+    classifierPort.connect(inputSignal);
+    const auto classifierSignal = classifierFb.getSignals()[0];
+
+    const auto classifierSignalDescription = classifierSignal.getDescriptor();
+    const auto classifierSignalLabelsSize = classifierSignal.getDescriptor().getDimensions()[0].getSize();
```

Comment:
There's a lot of duplicate variables here that use different names to describe the same thing. In this case

```
classifierSignalLabelsSize == classCount
```

This makes the test cases hard to read and to grasp what each variable represents.